src/hotspot/cpu/riscv/templateInterpreterGenerator_riscv.cpp

```diff
@@ -1097,8 +1097,12 @@ address TemplateInterpreterGenerator::generate_native_entry(bool synchronized) {
     Label L;
     __ ld(x28, Address(xmethod, Method::native_function_offset()));
     address unsatisfied = (SharedRuntime::native_method_throw_unsatisfied_link_error_entry());
-    __ mv(t1, unsatisfied);
-    __ ld(t1, Address(t1, 0));
+    __ mv(t, unsatisfied);
+    if (AvoidUnalignedAccesses) {
```

Comment:
Flag `AvoidUnalignedAccesses` is checked in function `load_long_misaligned`. No need for another check for the same flag here. So seems that the if-else structure here could be simplified into one single line:
```
__ load_long_misaligned(t1, Address(t, 0), t0, 2); // 2 bytes aligned, but not 4 or 8
```
Looks good otherwise. Thanks again.